### PR TITLE
Shift PostLabelingDelay(s) by slice times

### LINKED ISSUE #189: PLD/TI not adjusted by slice timing for 2D datasets?
I have a 2D PCASL dataset processed through ASLPrep. However, it looks like the slice-timing information is used to run AFNI's 3dTshift on the raw tag/control data, but not to adjust the PLDs/TIs for individual slices before performing CBF quantification. Is this reading of the current behavior correct?

On the same dataset, I observed lower CBF values from ASLprep compared to an in-house MATLAB script for quantifying CBF which accounted for slice-delay in the PLD values. On the other hand, when adjusting the MATLAB script to ignore slice-delay (i.e. use the PLD of the very first slice, for all slices), the CBF values were much more similar to ASLprep. ASLprep ran through to completion, and the report also indicated slice-timing correction was performed.

If this is the current behavior, it would be great to use the slice-delay in PLD/TI in CBF quantification per the white paper recommendation. 

If it is already supposed to be in there, then I'd appreciate any pointers on running it properly :)

EDIT: This was back when I was using 0.2.7. I have not had a chance to re-test with 0.2.8.

### PR DESCRIPTION
**Pull request by tsalo**
Closes #189.

## Changes proposed in this pull request

- Shift PLDs by slice times, when necessary.

### CODE REVIEW

File: aslprep/interfaces/cbf_computation.py [hunk 1]
```diff
@@ -528,10 +524,29 @@ def _run_interface(self, runtime):
...
+        if "SliceTiming" in metadata:
+            slice_times = metadata["SliceTiming"]
+            deltam_img = nb.load(deltam_file)
+            shape = deltam_img.shape[:3]
+
+            pld_brain = np.tile(plds, list(shape) + [1])
+
+            # Assumes XXI orientation?
+            pld_brain = pld_brain + slice_times[None, None, :, None]
+            pld_img = nb.Nifti1Image(pld_brain, deltam_img.affine, deltam_img.header)
```
Comment (tsalo) [inline aslprep/interfaces/cbf_computation.py]: @mattcieslak would you mind taking a look at this? I don't how to account for the image orientation and slice encoding direction (which is an optional metadata field) when shifting the values by the slice times.

**Comment by tsalo** [comment]
`scrub_summary` is failing for the Philips dataset (which is the only single-PLD test dataset with slice timing info).

**Comment by MartinGell** [comment]
Hey Taylor,

Dont mean to open up old things, but I have been running aslprep using a similar sequence as the author of the original issue: https://github.com/PennLINC/aslprep/issues/189 

When running aslprep I noticed this warning: 
260311-14:20:11,569 nipype.utils WARNING:
         Slice times are not ascending. They will be ignored in the BASIL call.
         
Going down a rabbit hole trying to figure out how to fix this I found the above mentioned issue and your solution to it. Looking at my *asl_DeltaMOrCBF_plds.nii.gz file in /compute_cbf I think I can verify that the fix you suggested has been implemented. So I'm wandering if maybe the WARNING is a leftover legacy warning as it makes it sound there will be no slice time correction of any kind when in fact your voxelwise solution is in fact implemented.

Hope this is useful in any way!

Martin

-----
Version: 26.0.0
Acquisition: 2D multiband pCASL, interleaved slice timing
asl_fit_wf contains no slice timing correction node

**Comment by tsalo** [comment]
That file is used for ASLPrep's internal CBF calculation (which does work for non-ascending slice acquisitions), but BASIL (the FSL tool for calculating CBF) uses a command-line argument to define the slice timing, and it just accepts a value for the time between slices (`--slicesdt` I think) that only works for ascending acquisitions.

**Pull request merged**